internal/repometadata/repometadata.go

```diff
@@ -0,0 +1,270 @@
+	IssueTracker string `json:"issue_tracker"`
+
+	// Language is the programming language (e.g., "rust", "python", "go").
+	Language string `json:"language,omitempty"`
+
+	// LibraryType is the type of library (e.g., "GAPIC_AUTO").
+	LibraryType string `json:"library_type,omitempty"`
+
+	// Name is the API short name.
+	Name string `json:"name,omitempty"`
+
+	// NamePretty is the human-readable name of the API.
+	NamePretty string `json:"name_pretty,omitempty"`
+
+	// ProductDocumentation is the URL to the product documentation.
+	ProductDocumentation string `json:"product_documentation,omitempty"`
+
+	// ReleaseLevel is the release level (e.g., "stable", "preview").
+	ReleaseLevel string `json:"release_level,omitempty"`
+
+	// Repo is the repository name (e.g., "googleapis/google-cloud-rust").
+	Repo string `json:"repo,omitempty"`
+}
+
+// GenerateRepoMetadata generates the .repo-metadata.json file by parsing the service YAML.
+func GenerateRepoMetadata(library *config.Library, language, repo, serviceConfigPath, outdir string, apiPaths []string) error {
+	svcCfg, err := serviceconfig.Read(serviceConfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to read service config: %w", err)
+	}
+
+	// Select the default version from all API paths, preferring stable versions,
+	// but with the ability to override.
+	defaultVersion := library.DefaultVersionOverride
+	if defaultVersion == "" {
+		defaultVersion = selectDefaultVersion(apiPaths)
+	}
+
+	// Build client documentation URL based on language
+	clientDocURL := buildClientDocURL(language, extractNameFromAPIID(svcCfg.GetName()))
+
+	// Create metadata
+	metadata := &RepoMetadata{
+		APIID:               svcCfg.GetName(),
+		NamePretty:          cleanTitle(svcCfg.GetTitle()),
+		ClientDocumentation: clientDocURL,
+		ReleaseLevel:        library.ReleaseLevel,
+		Language:            language,
+		LibraryType:         "GAPIC_AUTO",
+		Repo:                repo,
+		DistributionName:    library.Name,
+		DefaultVersion:      defaultVersion,
+	}
+
+	// Add optional fields if available
+	if svcCfg.GetPublishing() != nil {
+		publishing := svcCfg.GetPublishing()
+		if publishing.GetDocumentationUri() != "" {
+			metadata.ProductDocumentation = extractBaseProductURL(publishing.GetDocumentationUri())
+		}
+		if publishing.GetApiShortName() != "" {
+			metadata.APIShortname = publishing.GetApiShortName()
+			metadata.Name = publishing.GetApiShortName()
+		}
+	}
+
+	// Set API description from override or service YAML
+	if library.DescriptionOverride != "" {
+		// Use override from library configuration
+		metadata.APIDescription = library.DescriptionOverride
+	} else if svcCfg.GetDocumentation() != nil && svcCfg.GetDocumentation().GetSummary() != "" {
+		// Fall back to service YAML documentation
+		metadata.APIDescription = strings.TrimSpace(svcCfg.GetDocumentation().GetSummary())
+	}
+
+	// Set default release level if not specified
+	if metadata.ReleaseLevel == "" {
```

Comment:
I think you should be able to use library.ReleaseLevel and that will always be accurate